restli-tools/src/main/java/com/linkedin/restli/tools/symbol/SymbolTableNameHandler.java

```diff
@@ -56,26 +56,37 @@
    */
   String generateName(List<String> symbols)
   {
+    if (_serverNodeUri == null)
+    {
+      throw new IllegalStateException("Cannot generate symbol table name with null server node URI.");
+    }
+
     return _serverNodeUri + SERVER_NODE_URI_PREFIX_TABLENAME_SEPARATOR + _symbolTablePrefix
         + PREFIX_HASH_SEPARATOR + symbols.hashCode();
   }
 
   @Override
   protected SymbolTableMetadata createMetadata(String serverNodeUri, String tableName) {
     // A table is remote if the server node URI does not match the current server node URI.
-    boolean isRemote = !_serverNodeUri.equals(serverNodeUri);
+    boolean isRemote = _serverNodeUri == null || !_serverNodeUri.equals(serverNodeUri);
     return new SymbolTableMetadata(serverNodeUri, tableName, isRemote);
   }
 
   /**
-   * Rename the original table name, replacing the existing url prefix with this instance's url prefix.
+   * Rename the original table name, replacing the existing url prefix with this instance's url prefix if it exists. If
+   * this instance doesn't have a url prefix, then this method is a no-op and returns the table name as is.
    *
    * @param existingTableName The existing table name.
    *
    * @return The new name with the url prefix in the existing name replaced with this instance's url prefix.
    */
   String replaceServerNodeUri(String existingTableName)
```

Comment:
Not related to this change but why are we replacing remote uri with servers own URI? was the initial intention to combine downstream symbol tables?

Reply:
Ah the idea was that if you're caching it there is a possibility that in the future you may serve it from your own host, esp if its different versions of the same service etc. So we added it in as an additional precaution.